### PR TITLE
Fix: `Federated search/browse` results count by portal

### PR DESCRIPTION
**Pull request by Bilelkihal**
Related issue: https://github.com/ontoportal-lirmm/bioportal_web_ui/issues/859

### Changes
- Include counting duplicated results for each portal (not only the canonical) in both federated browse and search
(https://github.com/ontoportal-lirmm/bioportal_web_ui/pull/870/commits/9f0fee89da72ef8a9b0304a483039d10a62e6197) (https://github.com/ontoportal-lirmm/bioportal_web_ui/pull/870/commits/5f460ffc76249688fb4b0397e2997dc31cc2d2ec)

### Screenshots:
**Federated browse**
image

**Federated search**
image

### CODE REVIEW

**Review by Bilelkihal — COMMENTED** [review]
✅

File: app/helpers/federation_helper.rb [hunk 1]
```diff
@@ -234,7 +238,7 @@ def counts_ontology_ids_by_portal_name(portals_ids)
       counts[current_portal.downcase] += 1 if id.include?(current_portal.to_s.downcase)
 
       federation_portals.each do |portal|
-        portal_api = federated_portals[portal.downcase.to_sym][:api]
+        portal_api = federated_portals[portal.downcase.to_sym][:name].downcase
```
Comment (syphax-bouazzouni) [inline app/helpers/federation_helper.rb:241]: Using a portal's name does not always work. Sometimes, the API URL is not the same as the portal name. 
And what did you need to change this line?

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]

File: app/helpers/federation_helper.rb (hunk 1 above)
Comment (Bilelkihal) [reply app/helpers/federation_helper.rb:241]: In the search page, to count the results that are in the `:other_portals` list, we don't have the api url, but instead we have the ui url. 
Ok no big deal, to solve this I will just check both like this:
`counts[portal.downcase] += 1 if (id.include?(portal_api) || id.include?(portal_ui))`

File: app/helpers/federation_helper.rb [hunk 2]
```diff
@@ -234,8 +238,9 @@ def counts_ontology_ids_by_portal_name(portals_ids)
       counts[current_portal.downcase] += 1 if id.include?(current_portal.to_s.downcase)
 
       federation_portals.each do |portal|
-        portal_api = federated_portals[portal.downcase.to_sym][:api]
-        counts[portal.downcase] += 1 if id.include?(portal_api)
+        portal_api = federated_portals[portal.downcase.to_sym][:api].sub(/^https?:\/\//, '')
```
Comment (Bilelkihal) [inline app/helpers/federation_helper.rb:241]: @syphax-bouazzouni I added `sub(/^https?:\/\//, '')` here to make sure to deal with http, https variations.
cause I found for example the ontology STY in earthportal is using http `"http://earthportal.eu/ontologies/STY"` while the UI link is using https `"https://ecoportal.lifewatch.eu/"`

Comment (syphax-bouazzouni) [reply app/helpers/federation_helper.rb:241]: Yeah, sometimes you can even get, an IP, not the actual domain name. but anyway for now good enough.

**Review by syphax-bouazzouni — APPROVED** [review]

**Pull request merged**